tests/auth/ldap_test.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+"""Test ldap
+
+:copyright: Copyright (c) 2023 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+import pytest
+from pykern.pkdebug import pkdlog, pkdp
+
+
+def test_happy_path():
+    from pykern.pkcollections import PKDict
+    from pykern import pkunit
+
+    res = PKDict(
```

Comment:
`res` stands for result in our style. Some people use `rv` (return value) for the same thing. `res` here is therefore incorrect. Moreover, you don't need the temporary as it is only used once. 

Once you refactor that for all these methods, you will notice that _call_login could accept email, password and construct the PKDict.